pkg/teeth/publish_logs.go

```diff
@@ -89,3 +101,52 @@ func SetUpPublisherSourceQuery(ctx context.Context, bqClient BigQueryClient) (*b
 	}
 	return bqClient.Query(b.String()), nil
 }
+
+// SaveInvocationCommentStatus inserts the statuses into the
+// InvocationCommentStatus table. It uses the batching parameters
+// of the BigQuery client to configure batch inserts.
+//
+// Returns a list of errors where each error corresponds to the batch
+// insert in order.
+func SaveInvocationCommentStatus(ctx context.Context, bqClient BigQueryClient, statuses []*InvocationCommentStatusRecord) []error {
+	errors := make([]error, 0)
+	for i := 0; i <= len(statuses)/bqClient.InsertBatchSize(); i++ {
+		// Wait to prevent being rate-limited by BigQuery insert API.
+		if i > 0 {
+			time.Sleep(bqClient.InsertBatchWait())
```

Comment:
Could use `https://github.com/sethvargo/go-retry` instead of using `sleep` can be better?

Not a strong opinion, just pointing out a possible alternative.

Reply:
Cool thanks! I'll definitely need this eventually.

So yes, a backoff implementation may work ... but (1) could be less efficient because the quota problem occurs when BigQuery batch inserts are done within a very short period of time - thundering herd (2) would require more error handling. 

Example: Constant backoff of 1s
`t+0s`: N calls happen, N-1 calls get some quota error or 429 Too Many Requests, gotta handle user error and have N-1 calls wait 1s ...
`t+1s`: N-1 calls happen, N-2 calls get user error, handle user error and have N-2 calls wait 1s ...
etc. ... not-to-mention the cases when 1 or calls get some internal error.

I see that https://github.com/sethvargo/go-retry also supports random jitters and max retries to help fix these issues. But if this is already sufficient and simple, then I will play the YAGNI card :)

Reply:
That's a good point about context awareness. I added changes to handle cancellation + timeouts. PTAL.

Reply:
The sleep still doesn't handle cancellation / timeout properly. It would need something like:

```go
select {
case <-ctx.Done(): //context cancelled
case <-time.After(bqClient.InsertBatchWait()):
}
```

Using sethvargo/go-retry actually saves you the trouble to do this yourself.

Reply:
Thanks. Added.

> Using sethvargo/go-retry actually saves you the trouble to do this yourself.
Sure but IIUC it'd still be different from what's implemented here. There'd be concurrent calls with some random backoff in the range of [backoff-jitter, backoff+jitter]. Here the calls are serialized. 

Honestly, I don't see a serious need for the batching and was doing this for defensive programming reasons whereby 1000s of PRs got merged into a single repo. 

Given how high BigQuery's insert quota/limits are, I'm gonna remove it all. If there's a discussion on needing it,  I can stack it in another PR.

Reply:
The calls would be serialized with go-retry too.